pyclam/criterion.py

```diff
@@ -119,6 +141,90 @@ def __call__(self, root: Cluster) -> Set[Cluster]:
         return {cluster for cluster in root.manifold.layers[-1].clusters}
 
 
+class Layer(SelectionCriterion):
+    def __init__(self, depth: int):
+        if depth <= 0:
+            raise ValueError(f'expected a positive depth. got: {depth}')
+        self.depth = depth
+
+    def __call__(self, root: Cluster):
+        manifold: Manifold = root.manifold
+        if manifold.depth <= self.depth:
+            return [cluster for cluster in manifold.layers[-1].clusters]
+        else:
+            return [cluster for cluster in manifold.layers[self.depth].clusters]
+
+
+class SelectionClauses(SelectionCriterion):
+    def __init__(self, clauses: Iterable[Clause]):
+        if any((not isinstance(c, Clause) for c in clauses)):
+            raise TypeError(f'selection clauses mut all be of type Clause.')
```

Comment:
typo

Reply:
removed this decision-tree based criterion